packages/flutter/lib/src/material/arc.dart

```diff
@@ -201,7 +201,7 @@ T _maxBy<T>(Iterable<T> input, _KeyFunc<T> keyFunc) {
   dynamic maxKey;
   for (T value in input) {
     final dynamic key = keyFunc(value);
-    if (maxKey == null || key > maxKey) {
+    if (maxKey == null || (key > maxKey) as bool) {
```

Comment:
`_KeyFunc` is only used internally and only used by one caller that returns a `double`.  Can we just change that to 

```dart
typedef _KeyFunc<T> = double Function(T input);
```

And then drop the `dynamics` in this method and the cast?

Reply:
Good catch! Done.